command/agentproxyshared/auth/cert/cert_test.go

```diff
@@ -188,4 +195,104 @@ func TestCertAuthMethod_AuthClient_withCertsReload(t *testing.T) {
 	if reloadedClient == clientToUse {
 		t.Fatal("expected client from AuthClient to return back a new client")
 	}
+
+	method.CredSuccess()
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	inst := method.(*certMethod)
+	inst.clientCert = "./test-fixtures/keys/cert1.pem"
+	select {
+	case <-ctx.Done():
+	case <-method.NewCreds():
+		cancel()
+		t.Fatal("malformed cert should not be observed as a change")
+	}
+
+	inst.clientKey = "./test-fixtures/keys/key1.pem"
+	ctx, cancel = context.WithTimeout(context.Background(), 3*time.Second)
+	select {
+	case <-ctx.Done():
+		t.Fatal("failed to watch the cert change: timeout")
+	case <-method.NewCreds():
+		cancel()
+	}
+}
+
+func TestCertAuthMethod_hashCert_withEmptyPaths(t *testing.T) {
```

Comment:
Sorry I missed this in my last review, but these tests should have GoDocs (a comment above them starting with the test name), e.g.:

```
// TestCertAuthMethod_hashCert_withEmptyPaths is a test that...
```

Reply:
I added some doc elaborating the purposes, pls take a look